src/components/ruler/Arrows.tsx

```diff
@@ -2,26 +2,23 @@ import leftArrowIcon from "../../assets/icons/arrow-left.svg";
 import rightArrowIcon from "../../assets/icons/arrow-right.svg";
 
 interface IProps {
-  startIndex: number;
-  setStartIndex: (val: number) => void;
+  leftPosition: number;
+  setLeftPosition: (val: number) => void;
 }
-const Arrows = ({ startIndex, setStartIndex }: IProps) => {
+
+const Arrows = ({ leftPosition, setLeftPosition }: IProps) => {
   const handleArrowClick = (direction: "left" | "right") => {
-    const step = 1;
-    if (direction === "left") {
-      setStartIndex(Math.max(0, startIndex - step));
-    } else {
-      setStartIndex(Math.min(80, startIndex + step));
-    }
+    const step = window.innerWidth / 21;
```

Comment:
Put the numbers in variables, so they understand what they are
code the calculation 3.8

Reply:
also , put window.innerWidth in state and update the state every change of screen